storage-proofs/src/parameter_cache.rs

```diff
@@ -181,43 +179,45 @@ where
             .or_else(|_| write_cached_metadata(&meta_path, Self::cache_meta(pub_params)))
     }
 
-    fn get_groth_params<R: RngCore>(
-        rng: Option<&mut R>,
-        circuit: C,
-        pub_params: &P,
-    ) -> Result<groth16::Parameters<E>> {
+    fn get_groth_params(circuit: C, pub_params: &P) -> Result<groth16::MappedParameters<E>> {
         let id = Self::cache_identifier(pub_params);
 
+        // This is used for testing only, if parameters are otherwise unavailable.
         let generate = || -> Result<_> {
-            if let Some(rng) = rng {
-                info!("Actually generating groth params. (id: {})", &id);
-                let start = Instant::now();
-                let parameters = groth16::generate_random_parameters::<E, _, _>(circuit, rng)?;
-                let generation_time = start.elapsed();
-                info!(
-                    "groth_parameter_generation_time: {:?} (id: {})",
-                    generation_time, &id
-                );
-                Ok(parameters)
-            } else {
-                bail!("No cached parameters found for {}", id);
-            }
+            use rand::SeedableRng;
+            use rand_xorshift::XorShiftRng;
+            use std::time::Instant;
+
+            info!("Actually generating groth params. (id: {})", &id);
+            let rng = &mut XorShiftRng::from_seed(crate::TEST_SEED);
```

Comment:
Sorry I forgot this before, lets not use a seeded rng, but rather `thread_rng`, these should not be regeneratable.

Reply:
Ok, thanks, will update this shortly!